src/analysis/retail/monk/mistweaver/modules/spells/ShaohaosLessons.tsx

```diff
@@ -0,0 +1,164 @@
+import { formatNumber, formatPercentage } from 'common/format';
+import SPELLS from 'common/SPELLS';
+import { TALENTS_MONK } from 'common/TALENTS';
+import HIT_TYPES from 'game/HIT_TYPES';
+import { SpellIcon } from 'interface';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import {
+  calculateEffectiveDamage,
+  calculateEffectiveDamageReduction,
+  calculateEffectiveHealing,
+} from 'parser/core/EventCalculateLib';
+import Events, { DamageEvent, EventType, HealEvent } from 'parser/core/Events';
+import StatTracker from 'parser/shared/modules/StatTracker';
+import BoringValueText from 'parser/ui/BoringValueText';
+import ItemDamageDone from 'parser/ui/ItemDamageDone';
+import ItemHealingDone from 'parser/ui/ItemHealingDone';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+
+const DESPAIR_CRIT_INCREASE = 0.3;
+const DOUBT_INCREASE = 0.35;
+const FEAR_HASTE_INCREASE = 0.25;
+const FEAR_MITIGATION_PERCENT = 0.15;
+
+class ShaohaosLessons extends Analyzer {
+  static dependencies = {
+    statTracker: StatTracker,
+  };
+  buffCount: Map<number, number> = new Map<number, number>();
+  angerDamage: number = 0;
+  angerHealing: number = 0;
+  despairDamage: number = 0;
+  despairHealing: number = 0;
+  doubtDamage: number = 0;
+  doubtHealing: number = 0;
+  curHpPercent: number = 0;
+  fearMitigated: number = 0;
+  protected statTracker!: StatTracker;
+  constructor(options: Options) {
+    super(options);
+    this.active = this.selectedCombatant.hasTalent(TALENTS_MONK.SHAOHAOS_LESSONS_TALENT);
+    this.addEventListener(
+      Events.heal.by(SELECTED_PLAYER).spell(SPELLS.LESSON_OF_ANGER_HEAL),
+      this.handleHealAnger,
+    );
+    this.addEventListener(
+      Events.damage.by(SELECTED_PLAYER).spell(SPELLS.LESSON_OF_ANGER_DAMAGE),
+      this.handleDamageAnger,
+    );
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER), this.handleHealDoubt);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER), this.handleDamageDoubt);
+    this.addEventListener(Events.heal.by(SELECTED_PLAYER), this.handleDespair);
+    this.addEventListener(Events.damage.by(SELECTED_PLAYER), this.handleDespair);
+    this.addEventListener(Events.heal.to(SELECTED_PLAYER), this.healTaken);
+    this.addEventListener(Events.damage.to(SELECTED_PLAYER), this.damageTaken);
+  }
+
+  healTaken(event: HealEvent) {
```

Comment:
You can also  add a cast event listener to increase the accuracy of health tracking here